users/forms.py

```diff
@@ -0,0 +1,28 @@
+from django import forms
+from .models import User
+
+
+class UserCreationForm(forms.ModelForm):
+    password1 = forms.CharField(label="Password", widget=forms.PasswordInput())
+    password2 = forms.CharField(label="Password confirmation", widget=forms.PasswordInput())
+
+    class Meta:
+        model = User
+        fields = ('email', 'first_name', 'last_name', 'birth_date',)
+
+    def clean_password2(self):
+        password1 = self.cleaned_data.get('password1')
+        password2 = self.cleaned_data.get('password2')
+
+        if not password2:
+            raise forms.ValidationError('You must confirm your password')
+        if password1 != password2:
+            raise forms.ValidationError('Your passwords do not match')
+        return password2
+
+    def save(self, commit=True):
+        new_user = super(UserCreationForm, self).save(commit=False)
+        new_user.set_password(self.cleaned_data["password1"])
+        if commit:
```

Comment:
What's this 'commit'?

Reply:
Why would you call save if you didn't want to save it?

Reply:
When you use save with commit=False it returns an instance of the model the form is attached to. It just doesn't save it to the data base right away. So if you want to modify the instance more after the form is done, you can